wordpress-mobile/WordPress-iOS
# Prime for Push: Winback

Fixes #9185

For our existing users that denied push notifications before prime for push was added to the app, the winback gives them a second chance to enable them with an inline alert on the notifications tab. This is the final remaining feature of Prime for Push.

![winback](https://user-images.githubusercontent.com/517257/40552274-6ea604dc-5ffc-11e8-9a4e-d2a20baef6b2.png)


To test:
> Testing is easiest by commenting out one line in `BlogDetailsViewController.m` on line `1170`:
```swift
//[NSUserDefaults standardUserDefaults].notificationPrimerAlertWasDisplayed = YES;
```
1. Change code as above
2. Delete the app and reinstall
3. Login to a wpcom account
4. When the prime for push alert appears, tap Allow
5. When the iOS alert appears, tap "Don't Allow" (you traitor)
6. Go to the notifications tab
7. Verify that the winback appears, and works
